org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/publisher/TranslationImplementation.java

```diff
@@ -21,40 +21,56 @@
  */
 
 
-import java.util.Date;
-import java.util.Map;
-import java.util.Set;
-
+import org.apache.commons.lang3.StringUtils;
 import org.hl7.fhir.r5.model.Enumeration;
 import org.hl7.fhir.r5.model.PrimitiveType;
 import org.hl7.fhir.r5.utils.TranslatingUtilities.TranslationServices;
 
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
 public class TranslationImplementation implements TranslationServices {
 
   // marker is used to see what content is not being translated (debugging)
   private static final String MARKER1 = ""; // "^";
   private static final String MARKER2 = ""; // "^^";
-  
+  private Map<String, String> keys = new HashMap<>();
   // -- configuration -------------------------------------------------------
   
   private String lang; 
-  
+  private String nonAlpha = "[^a-zA-Z0-9_]+";
+  private String noEndDash = "_$";
+  private Locale locale = Locale.getDefault();
+
+  private ResourceBundle bundle = ResourceBundle.getBundle("Messages");
+  private boolean harvest = false;
+
   public String getLang() {
     return lang;
   }
 
   public void setLang(String lang) {
-    this.lang = lang;
+    if("harvest".equals(lang)) {
+      this.harvest = true;
+    } else {
+      this.lang = lang;
+      this.locale = Locale.forLanguageTag(lang);
+      bundle = ResourceBundle.getBundle("Messages", this.locale);
```

Comment:
Put any magic numbers or strings into final variables with commented explanations of what their purpose is please.